src/nspawn/nspawn-mount.c

```diff
@@ -83,13 +83,22 @@ static int custom_mount_compare(const CustomMount *a, const CustomMount *b) {
         return CMP(a->type, b->type);
 }
 
-static bool source_path_is_valid(const char *p) {
+static int source_path_parse(const char *p, bool allow_empty, char **ret) {
```

Comment:
allow_empty sounds if you'd just allow empty strings. But you do more than that.. you convert them into NULL...

i am not sure this should be a bool param. instead, just drop it here entirely, but add a new helper func: source_path_parse_nullable() that wraps source_path_path(), but converts empty string to NULL too.